Autopilot/Tools/build.bash

```diff
@@ -89,6 +94,66 @@ if [[ $RUN_UNIT_TESTS == true ]]; then
         ./$SCRIPT;
     done
 
+elif [[ $RUN_SIMULATION == true ]]; then
+
+    echo "Building Simulation !"
+    echo ""
+    echo ""
+
+    BUILD_DIR="SimulationBuild"
+
+    if [[ $CLEAN == true ]]; then
+        echo "Cleaning old build environment"
+        cmake -E remove_directory $BUILD_DIR
+        cmake -E make_directory $BUILD_DIR
+    fi
+
+    # we only want to run the simulation if things have changed. To check this, we will check if the time stamp of the exe is older than the time stamp at time of build.
+    TIME_OF_BUILD=$(date +"%s")
+
+    cmake -E chdir $BUILD_DIR\
+      cmake \
+        -G "${GENERATOR}" \
+        -D KIND_OF_BUILD="SIMULATION"\
+        -Wdev\
+        -Wdeprecated\
+        ../
+    cmake --build $BUILD_DIR
+
+
+    AGE_OF_EXE=$(date -r $BUILD_DIR/sim +"%s")
+
+    if [[ $AGE_OF_EXE -gt $TIME_OF_BUILD ]]; then
```

Comment:
Kinda curious to know what `-gt` means. 

I assume it checks if `$AGE_OF_EXE` is less than `$TIME_OF_BUILD` so we reset the simulation results.

Reply:
-gt means "greater than", not sure why bash doesn't just use " > ". yes your'e right. The point is that if the exe is newer than the time the build was started, then it was rebuilt by cmake (cmake doesent rebuild if no source files change) and it's necessary to run the simulation again.